recipes/master_config_post.rb

```diff
@@ -73,78 +79,78 @@
   persistent_storage node['cookbook-openshift3']['persistent_storage']
 end
 
-execute "Wait up to 60s for nodes registration [Size : #{node_servers.size.to_i}]" do
-  command "[[ `oc get node --no-headers --config=admin.kubeconfig 2> /dev/null | wc -l` -eq #{node_servers.size.to_i} ]]"
-  cwd node['cookbook-openshift3']['openshift_master_config_dir']
-  only_if "[[ `oc get node --no-headers --config=admin.kubeconfig 2> /dev/null | wc -l` -ne #{node_servers.size.to_i} ]]"
-  retries 12
-  retry_delay 5
+execute "Wait up to 30s for node registrations (Best Effort) [Expected number of Nodes: #{node_servers.size.to_i}]" do
+   command "for ((i = 0 ; i < 6 ; i++)); do [[ `oc get node --no-headers --config=admin.kubeconfig 2> /dev/null | wc -l` -eq #{node_servers.size.to_i} ]] && break || sleep 5; done"
+   cwd node['cookbook-openshift3']['openshift_master_config_dir']
+   only_if "[[ `oc get node --no-headers --config=admin.kubeconfig 2> /dev/null | wc -l` -ne #{node_servers.size.to_i} ]]"
 end
 
-node_servers.reject { |h| h.key?('skip_run') }.each do |nodes|
-  execute "Set schedulability for Master node : #{nodes['fqdn']}" do
-    command "#{node['cookbook-openshift3']['openshift_common_admin_binary']} manage-node #{nodes['fqdn']} --schedulable=${schedulability} --config=admin.kubeconfig"
-    environment(
-      'schedulability' => !nodes.key?(:schedulable) && master_servers.find { |server_node| server_node['fqdn'] == nodes['fqdn'] } ? 'False' : nodes['schedulable'].to_s
-    )
-    cwd node['cookbook-openshift3']['openshift_master_config_dir']
-    only_if do
-      master_servers.find { |server_node| server_node['fqdn'] == nodes['fqdn'] } &&
-        !Mixlib::ShellOut.new("oc get node | grep #{nodes['fqdn']}").run_command.error?
+if `[[ $(oc get node --no-headers --config=#{node['cookbook-openshift3']['openshift_master_config_dir']}/admin.kubeconfig 2> /dev/null | wc -l) -ge 1 ]]`
```

Comment:
This line makes the cookbook less reproductible. In the case where the above execute resource ("Wait up to 30s for node registration (Best Effort)") exists without at least one node being detected, then the registry/router/metrics won't be deployed and the kitchen test will fail. Note: the loop exiting with no node being detected is the case that happens on the first run of "kitchen converge" for all "standalone" variants.

I believe that registry/router/metrics should be deployed regardless of nodes being available. If nodes are not available, then the pods created by the registry/router/metrics LWRP just won't schedule.

Please consider removing that "if" and the associated "end".